File: backend/src/main/java/com/ubre/backend/dto/RideProgressDto.java
package com.ubre.backend.dto;

// Ride progress updates, including current location and estimated time of arrival to destination

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
public class RideProgressDto {
    private Long rideId;
    private WaypointDto location;
    private Integer estimatedTime;

    public RideProgressDto(Long rideId, WaypointDto location, int estimatedTime) {
        this.rideId = rideId;
        this.location = location;
        this.estimatedTime = estimatedTime;
    }
}

File: frontend/ubre/src/app/dtos/ride-progress-dto.ts
import { WaypointDto } from "./waypoint-dto";

export class RideProgressDto {
    public rideId : number;
    public location : WaypointDto;
    public estimatedTime : number;

    constructor(rideId : number, location : WaypointDto, estimatedTime : number) {
        this.rideId = rideId;
        this.location = location;
        this.estimatedTime = estimatedTime;
    }
}